OneCommunityGlobal/HighestGoodNetworkApp
# Jianjun - fix the permission of editing or deleting time entry from other users

# Description
**Fix the following two permission:**
 '**Edit Timelog Information**' : 'Gives the user permission to edit any time log entry.'
'**Delete Time Entry (Others)**' : 'Gives the user permission to Delete time entry from others users "Dashboard" -> "Leaderboard" -> "Dot By the side of user\'s name" -> "Current Time Log" -> "Trash button on bottom right"',

**Original problem description:**
7. (PRIORITY MEDIUM) Jae: (WIP Jianjun)Fix the “Edit Timelog Information” permission from the Manage User Permissions Component on Permission Management Page. Even after adding the permission, the user it’s not able to modify the information of the time entry from another user.
Admin/Owner login → Other Links → Permissions Management → Manage User’s Permissions → Choose User (must be a class other than Admin/Owner) → Edit Timelog Information → Add
Go log in as chosen user and test they have the new ability 
8. (PRIORITY MEDIUM) Jae: (WIP Jianjun)Fix the “Delete Time Entry (Others)” permission from the Manage User Permissions Component on Permission Management Page. Even after adding the permission, the user it’s not able to delete the time entry of another user.
Admin/Owner login → Other Links → Permissions Management → Manage User’s timer
Permissions → Choose User (must be a class other than Admin/Owner) → Delete Time Entry (Others) → Add
Go log in as chosen user and test they have the new ability 

## Main changes explained:
- The user with (front end) permission 'editTimelogInfo' is able to edit other user's time entry from that user's dashboard timelog tab.
- The user with (front end) permission 'deleteTimeEntryOthers' is able to delete other user's time entry from that user's dashboard timelog tab.
- Remove the unnecessary 'edit' button from the time log information(description/explanation) card.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as **user that is not Admin/Owner** --> "Leaderboard" -> "Dot By the side of other user\'s name" -> "Current Time Log" -> CANNOT see "Trash button on bottom right" or "Pen button on bottom right"
4. log as **Owner** --> Permissions Management → Manage User’s Permissions → Choose User (must be a class other than Admin/Owner)  → Delete Time Entry (Others) / Edit Timelog Information→ Add
5. log as **the user whose permission has been changed in step4**, repeat step 3, CAN SEE and CLICK the "Trash button" or "Pen button".

## Screenshots or videos of changes:
User without permission:
<img width="573" alt="1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/b9e3573d-5c96-4276-9102-719b3155f9f3">
Owner give permission to a certain user:
<img width="746" alt="1.5" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/647162ad-17d9-4869-99f3-bfa298f8ff31">
After getting the permission:
<img width="1083" alt="2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/cbf84593-96c9-4a5d-be4e-64302356f8ca">


## Note:
The permission 'editTimelogInfo' was not used as it designed in the previous code, and the "Edit" button of the info card has no action linked, so it is removed to avoid confusion.
